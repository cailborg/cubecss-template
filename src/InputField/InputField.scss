.textfield {
    & > .MuiInputBase-root {
        color: rgba(var(--lp-colors-neutral-800), 1);
    }
    & > .MuiFormLabel-root {
        color: rgba(var(--lp-colors-neutral-600), 1);
        font-size: var(--lp-font-size-body-2);
        font-family: 'Source Sans Pro', sans-serif;
        &.Mui-disabled {
            color: rgba(var(--lp-colors-neutral-400), 1);
        }
        &.Mui-error {
            color: rgba(var(--lp-colors-error-600), 1);
        }
        &.Mui-focused {
          color: rgba(var(--lp-colors-amethyst-600), 1);
      }
    }
    & > .MuiInput-underline:before {
        border-bottom: var(--lp-border-widths-thin) var(--lp-border-styles-solid) rgba(var(--lp-colors-neutral-300), 1);
    }
    & > .MuiInput-underline:after {
        border-bottom: var(--lp-border-widths-thick) var(--lp-border-styles-solid) rgba(var(--lp-colors-amethyst-600), 1);
    }
    & > .MuiInput-underline.Mui-error:after {
        border-bottom: var(--lp-border-widths-thick) var(--lp-border-styles-solid) rgba(var(--lp-colors-error-600), 1);
    }
    & > .MuiInput-underline:hover:not(.Mui-disabled):before {
        border-bottom: var(--lp-border-widths-thick) var(--lp-border-styles-solid) rgba(var(--lp-colors-neutral-300), 1);
    }
    & > .MuiInputBase-input {
        color: rgba(var(--lp-colors-neutral-800), 1);
    }
    & > .MuiFormHelperText-root {
        color: rgba(var(--lp-colors-neutral-600), 1);
        &.Mui-error {
            color: rgba(var(--lp-colors-error-600), 1);
        }
    }
}
