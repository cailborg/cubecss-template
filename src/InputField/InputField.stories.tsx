import React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";
import { InputField, InputFieldProps } from "./InputField";

export default {
  title: "Components/InputField",
  component: InputField,
  argTypes: {
    children: { control: "text" },
  },
} as Meta;

const Template: Story<InputFieldProps> = (args) => <InputField {...args} />;

export const Standard = Template.bind({});
Standard.args = {
  label: "TextField Label",
};
