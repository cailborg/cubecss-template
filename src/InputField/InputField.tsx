import React from "react";
import "./InputField.scss";
import TextField from "@material-ui/core/TextField";
import { InputModeType } from "./types";

export interface InputFieldProps {
    type?: string;
    ref?: React.Ref<HTMLElement>;
    inputRef?: React.Ref<HTMLInputElement>;
    label?: string;
    placeholder?: string;
    value?: string | null;
    defaultValue?: string;
    shrinkLabel?: boolean;
    maxLength?: number;
    disabled?: boolean;
    error?: string;
    helperText?: string;
    autoComplete?: string;
    trackingId?: string;
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
    onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
    onMouseDown?: (event: React.MouseEvent<HTMLInputElement>) => void;
    onMouseOver?: (event: React.MouseEvent<HTMLInputElement>) => void;
    onMouseLeave?: (event: React.MouseEvent<HTMLInputElement>) => void;
    onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
    onKeyUp?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
    onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
    inputMode?: InputModeType;
}

export const InputField: React.FC<InputFieldProps> = React.forwardRef(
    function InputField(
        {
            type,
            label,
            placeholder,
            value,
            defaultValue,
            maxLength,
            disabled,
            error,
            helperText: helperTextProp,
            autoComplete,
            trackingId,
            onChange,
            onFocus,
            onBlur,
            onMouseDown,
            onMouseOver,
            onMouseLeave,
            onKeyDown,
            onKeyUp,
            onKeyPress,
            inputRef,
            inputMode,
        },
        ref?: React.Ref<HTMLElement>
    ) {
      const helperText = error || helperTextProp;
        return (
            <TextField
                label={label}
                className="textfield"
                helperText={helperText}
                disabled={disabled}
                type={type}
                error={Boolean(error)}
                placeholder={placeholder}
                value={value}
                defaultValue={defaultValue}
                inputProps={{
                    inputRef,
                    autoComplete,
                    maxLength,
                    trackingId,
                }}
                onChange={onChange}
                onBlur={onBlur}
                onFocus={onFocus}
                onMouseDown={onMouseDown}
                onMouseOver={onMouseOver}
                onMouseLeave={onMouseLeave}
                onKeyDown={onKeyDown}
                onKeyUp={onKeyUp}
                onKeyPress={onKeyPress}
                inputMode={inputMode}
            />
        );
    }
);
