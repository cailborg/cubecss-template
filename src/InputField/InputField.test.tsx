import React from "react";
import { InputField } from "./InputField";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<InputField/>).toJSON();
  expect(tree).toMatchSnapshot();
});
