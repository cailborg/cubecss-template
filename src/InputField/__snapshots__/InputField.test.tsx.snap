// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root textfield"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
